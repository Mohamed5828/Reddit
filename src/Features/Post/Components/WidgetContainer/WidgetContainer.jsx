import {
  BodyContainer,
  HeaderContainer,
  InnerContainer,
  InnerHeaderContainer,
  MainContainer
} from "./WidgetContainer.styled";

/**
 * container of widgets in subreddit
 * 
 * @param {string} headerText - text in header of widget
 * @param {React.Component} children - component inside this component
 * @returns {React.Component} CreatePost component
 */
const WidgetContainer = ({ headerText, children }) => {


  /**
   * header of widget
   * 
   * @param {string} text - text pf widget header
   * @returns {React.Component} CreatePost component
   */
  const Header = ({ text }) => {
    return (
      <HeaderContainer>
        <InnerHeaderContainer>
          <h2>{text}</h2>
        </InnerHeaderContainer>
      </HeaderContainer>
    );
  };


  return (
    <MainContainer>
      <InnerContainer>
        <Header headerText={headerText} />
        <BodyContainer>{children}</BodyContainer>
      </InnerContainer>
    </MainContainer>
  );
};

export default WidgetContainer;
