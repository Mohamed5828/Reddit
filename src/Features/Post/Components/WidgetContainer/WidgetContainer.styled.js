import styled from "styled-components";

export const MainContainer = styled.div`
  margin-top: 16px;
  width: 312px;
`;

export const InnerContainer = styled.div`
  background-color: ${({theme})=>theme.background.primary} ;
  border: 1px solid ${({theme})=>theme.borderColor.secondary};
  border-radius: 4px;
  overflow: visible;
  word-wrap: break-word;
`;

export const HeaderContainer = styled.div`
  background-color: ${({theme})=>theme.color.secondary};
  fill: ${({theme})=>theme.background.primary};
  color: ${({theme})=>theme.background.primary};
  font-size: 10px;
  font-weight: 700;
  letter-spacing: 0.5px;
  line-height: 12px;
  text-transform: uppercase;
  border-radius: 3px 3px 0 0;
  padding: 0 12px 12px;
`;

export const InnerHeaderContainer = styled.div`
  padding: 12px 0 0;

  h2 {
    display: inline;
    font-size: 14px;
    line-height: 18px;
    font-weight: 700;
    text-transform: none;
  }
`;

export const BodyContainer = styled.div`
  padding: 12px;
  max-height: none;
`;
