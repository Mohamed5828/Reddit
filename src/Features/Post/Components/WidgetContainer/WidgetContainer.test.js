import Enzyme, { shallow } from "enzyme";
import WidgetContainer from "./WidgetContainer";

describe("container for widgets", () => {
  it("this is a test for Community Container", () => {
    expect(shallow(<WidgetContainer />)).toMatchSnapshot();
  });
});
