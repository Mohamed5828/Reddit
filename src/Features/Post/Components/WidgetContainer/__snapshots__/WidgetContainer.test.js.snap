// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container for widgets this is a test for Community Container 1`] = `
<styled.div>
  <styled.div>
    <Header />
    <styled.div />
  </styled.div>
</styled.div>
`;
